### ISSUE TITLE
error on npm install

### ISSUE THREAD

**Issue by arfat33** [angular/angular-cli#8612]
[2017-11-23T13_25_27_656Z-debug.log](https://github.com/angular/angular-cli/files/1499291/2017-11-23T13_25_27_656Z-debug.log)
Hi, I am trying to do 'npm install' inside one of my folders but i am getting below error:

npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: 403 Forbidden: https://registry.npmjs.org/fsevents/-/fsevents-1.1.3.tgz

**npm ERR! code E403
npm ERR! 403 Forbidden: binary-extensions@^1.0.0**

-------------------------------------
My npm version is 5.5.1
My node version is v8.9.1

Also my system was formatted today and I had to re-install and re-setup everything from scratch. npm install was working till yesterday before my system was formatted.

I have attached the log file here.

Thanks in advance

EDIT: every npm install is failing.
for reference here are the entries in my .npmrc file:
registry=http://registry.npmjs.org/
proxy=http://user_name:password@proxy:port
https-proxy=http://user_name:password@proxy:port
strict-ssl=false

I have tried many commands to get this working but no luck.
Note: my password contains '@' symbol but i have encoded it in my .npmrc file as %40

**Comment by filipesilva** [comment]
This seems like a problem with your node/npm and not with Angular CLI. 

Please have a look at the [fixing npm permissions page](https://docs.npmjs.com/getting-started/fixing-npm-permissions), [common errors page](https://docs.npmjs.com/troubleshooting/common-errors), [npm issue tracker](https://github.com/npm/npm/issues), or open a new issue if the problem you are experiencing isn't known.

**Comment by arfat33** [comment]
Thanks @filipesilva , the problem was with the registry. I changed http to https in registry and this error was gone

**Comment by m2art** [comment]
Solve problem this code:
`npm install --no-optional`

Check this link:
[https://github.com/npm/npm/issues/11632](https://github.com/npm/npm/issues/11632)
[https://stackoverflow.com/questions/46929196/how-to-solve-npm-install-throwing-fsevents-warning-on-non-mac-os](https://stackoverflow.com/questions/46929196/how-to-solve-npm-install-throwing-fsevents-warning-on-non-mac-os)

**Comment by i-Apoorva** [comment]
Use command: **npm install npm@latest -g**

then try again.
**Worked for me**

**Comment by AteeSS** [comment]
installing the older version of node i.e 8.9.4 worked for me.

**Comment by theshreyansh** [comment]
The thing that worked for me as below,
Issue: Registry defined at "C:\Users\*user-name*\.npmrc" was wrong. Need to update npm config registry.

Step 1
Solution: Removed registry entry from "C:\Users\*user-name*\.npmrc" file.

Step 2
After Step 1 I got below error 
>npm install -g @angular/cli
>npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning SELF_SIGNED_CERT_IN_CHAIN: request to https://registry.npmjs.org/@angular%2fcli failed, reason: self signed certificate in certificate chain
>npm WARN registry Using stale package data from https://registry.npmjs.org/ due to a request error during revalidation.

Issue: The npm maintainers announced on February 27th that npm’s Self-Signed Certificate is No More:
A bunch of users received a “SELFSIGNEDCERTINCHAIN” error during installing and publishing packages throughout the day today. npm no longer supports its self-signed certificates.
However, the recommended fix failed for me.

Solution: Open command prompt, and run below commands.
SETLOCAL
SET npm_config_strict-ssl=false
npm install npm -g --ca=null
npm --version
ENDLOCAL

Step 3 
Now >npm install -g @angular/cli running all ok.
